typescript/stepfunction-external-definition/lib/stepfunction-external-definition-stack.ts

```diff
@@ -0,0 +1,51 @@
+import * as cdk from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as apigateway from "aws-cdk-lib/aws-apigateway";
+import * as stepfunctions from "aws-cdk-lib/aws-stepfunctions";
+import * as logs from 'aws-cdk-lib/aws-logs';
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as fs from "fs";
+
+export class StepfunctionExternalDefinitionStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const logGroup = new logs.LogGroup(this, 'MyStateMachineLogGroup', {
+      logGroupName: '/aws/vendedlogs/states/MyStateMachine',
+      retention: logs.RetentionDays.ONE_WEEK, 
+    });
+    
+    const workflow = fs.readFileSync('./workflow/stepfunction.json.asl', 'utf8');
+    const stateMachine= new stepfunctions.StateMachine(this, 'MyStateMachine', {
+      stateMachineType: stepfunctions.StateMachineType.EXPRESS,
+      definitionBody: stepfunctions.DefinitionBody.fromString(workflow.toString()),
+      logs: {
+        destination: logGroup,
+        level: stepfunctions.LogLevel.ALL,
+        includeExecutionData: true,
+      },
+    });
+    stateMachine.addToRolePolicy(new iam.PolicyStatement({
```

Comment:
This appears to want to give access for the state machine to be able to write to the log group. Instead of creating a permission policy with a * in and having to track that you have all the correct API's, please change to using [LogGroup.granteWrite](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_logs.LogGroup.html#grantwbrwritegrantee).

It would look something like `logGroup.grantWrite(stateMachine.role);`